File: root/branches/will/junit/source/modules/strategy/src/test/sample_data/inputs/send_event.rb
include_class "org.marketcetera.strategy.ruby.Strategy"
include_class "java.lang.Long"

class SendEvent < Strategy
    def on_start
        @askCounter = 0
    end
    def on_other (data)
        do_send_event data
    end
    def on_callback(data)
        if(get_property("shouldRequestCEPData") != nil)
            do_cep_request
        else
            cancel_all_data_requests
        end
    end
    def on_ask(ask)
        @askCounter += 1
        set_property "askCount", Long.toString(@askCounter)
        set_property "ask", ask.toString
    end
    def do_send_event (event)
        if(get_property("eventOnlyTest") != nil)
            if(get_property("nilEvent") != nil)
                send_event nil
                return
            end
            send_event event
        else
            source = get_property "source"
            if(get_property("nilSource") != nil)
                send_event_to_cep event, nil  
                return
            end
            if(get_property("nilEvent") != nil)
                send_event_to_cep nil, source
                return
            end
            send_event_to_cep event, source
        end
    end
    def do_cep_request
        cepDataSource = get_property "source"
        if(cepDataSource != nil)
            statementString = get_property "statements"
            if(statementString != nil) 
                statements = statementString.split("#")
            else
                statements = nil
            end
            set_property("requestID", Long.toString(request_cep_data(statements.to_java(:string), cepDataSource)))
        end
    end
end

File: root/branches/1.1.x/source/modules/strategy/src/test/sample_data/inputs/SendEvent.java
import org.marketcetera.event.AskEvent;
import org.marketcetera.event.EventBase;
import org.marketcetera.strategy.java.Strategy;

/* $License$ */

/**
 * Sample strategy which tests the ability to send events to CEP and event subscribers.
 *
 * @author <a href="mailto:colin@marketcetera.com">Colin DuPlantis</a>
 * @version $Id$
 * @since 1.0.0
 */
public class SendEvent
        extends Strategy
{
    private int askCounter = 0;
    /* (non-Javadoc)
     * @see org.marketcetera.strategy.java.Strategy#onAsk(org.marketcetera.event.AskEvent)
     */
    @Override
    public void onAsk(AskEvent inAsk)
    {
        askCounter += 1;
        setProperty("askCount",
                    Integer.toString(askCounter));
        setProperty("ask",
                    inAsk.toString());
    }
    /* (non-Javadoc)
     * @see org.marketcetera.strategy.java.Strategy#onCallback(java.lang.Object)
     */
    @Override
    public void onCallback(Object inData)
    {
        if(getProperty("shouldRequestCEPData") != null) {
            doCepRequest();
        } else {
            cancelAllDataRequests();
        }
    }
    /* (non-Javadoc)
     * @see org.marketcetera.strategy.java.Strategy#onOther(java.lang.Object)
     */
    @Override
    public void onOther(Object inEvent)
    {
        doSendEvent((EventBase)inEvent);
    }
    /**
     * Sends the given event.
     *
     * @param inEvent an <code>EventBase</code> value
     */
    private void doSendEvent(EventBase inEvent)
    {
        if(getProperty("eventOnlyTest") != null) {
            if(getProperty("nilEvent") != null) {
                sendEvent(null);
                return;
            }
            sendEvent(inEvent);
        } else {
            String source = getProperty("source");
            if(getProperty("nilSource") != null) {
                sendEventToCEP(inEvent,
                               null);  
                return;
            }
            if(getProperty("nilEvent") != null) {
                sendEventToCEP(null,
                               source);
                return;
            }
            sendEventToCEP(inEvent,
                           source);
        }
    }
    /**
     * Sets up a CEP request.
     */
    private void doCepRequest()
    {
        String cepDataSource = getProperty("source");
        String statementString = null;
        if(cepDataSource != null) {
            statementString = getProperty("statements");
        }
        String[] statements;
        if(statementString != null) { 
            statements = statementString.split("#");
        } else { 
            statements = null;
        }
        setProperty("requestID",
                    Long.toString(requestCEPData(statements,
                                                 cepDataSource)));
    }
}
